CORL-753 | DOCs Updates (#2776)
coralproject/talk: docs/source/06-01-migrating-5.md

---

> title: Migrating to v5.0+

Reviewer: ```suggestion
title: Migrating to v5
```

permalink: /migration/5.0/
---

## **⏰It is currently not possible to upgrade directly from 4.x.x versions to version 5.x** ⏰

**Trying will break your install of Coral!**

Coral version 5.0.0 is more than just a version bump, it includes a fundamental rewrite of underlying core components, changes to the database schema, and the introduction of Typescript strong data types. This means migrating is a bit more complicated than running a simple database migration. 

### Migration Path
A step by step migration path is currently in development, and we will be announcing it here once it’s ready. 

The process for migration will include:
* Create a mongodump export of your 4.x.x database
* Use an import tool to transform the data for v5, and import it to a new mongo database
* Deploy & configure v5 on your production infrastructure (pointing to new DB)
* Update your embed script to the new version

> Downtime will likely be required to complete the migration, and you should carefully plan your migration accordingly depending on your infrastructure setup, and amount of data to be migrated. 

Reviewer: ```suggestion
Downtime will be required to complete the migration, and you should carefully plan your migration accordingly depending on your infrastructure setup, and amount of data to be migrated. 
```

### Migrating Plugins

> Plugins are no longer supported and will not work with any version 5.0+. Most of the optional `talk-plugin` features have been built right into core so you no longer need install, manage, and worry about plugins. If you were using plugins to achieve custom functionality, please contact us to learn about some of the new ways to solve custom use cases in version 5 and beyond. 

Reviewer: ```suggestion
Plugins are no longer supported and will not work with any instance of Coral >= v5. Most of the optional `talk-plugin` features have been built right into core so you no longer need install, manage, and worry about plugins. If you were using plugins to achieve custom functionality, please contact us to learn about some of the new ways to solve custom use cases in v5 and beyond. 
```
